scripts/macros/spells/spiritOfDeath.js

```diff
@@ -0,0 +1,221 @@
+    if (workflow.targets.size !== 1 || workflow.advantage) return;
+    let effect = effectUtils.getEffectByIdentifier(workflow.actor, 'spiritOfDeathHauntCreature');
+    if (!effect) {
+        workflow.aborted = true;   
+        return;
+    }
+    let {targets: validTargetUuids, formula} = effect.flags['chris-premades'].spiritOfDeathHauntCreature;
+    if (!validTargetUuids.includes(workflow.targets.first().document.uuid)) {
+        workflow.aborted = true;
+        return;
+    }
+    workflow.advantage = true;
+    workflow.attackAdvAttribution.add(genericUtils.translate('DND5E.Advantage') + ': ' + effect.name);
+}
+async function late({workflow}) {
+    let effect = effectUtils.getEffectByIdentifier(workflow.actor, 'summonedEffect');
+    let originItem = fromUuidSync(effect?.origin);
+    let sourceActor = originItem?.actor;
+    if(!sourceActor)
+    {
+        return;
+    }
+    let featureData = await compendiumUtils.getItemFromCompendium(constants.packs.summonFeatures, 'Haunt Creature: Haunt', {object: true, getDescription: true, translate: 'CHRISPREMADES.Macros.SpiritOfDeath.HauntCreatureHaunt', identifier: 'spiritOfDeathHauntCreatureHaunt', flatDC: sourceActor.system.attributes.spelldc});
+    if (!featureData) {
+        errors.missingPackItem();
+        return;
+    }
+    let targetEffectData = {
+        name: genericUtils.translate('CHRISPREMADES.Macros.SpiritOfDeath.Haunted'),
+        img: workflow.item.img,
+        origin: workflow.item.uuid,
+        flags: {
+            dae: {
+                "specialDuration": [
+                    "zeroHP"
+                ]
+            }
+        }
+    };
+    let casterEffectData = {
+        name: workflow.item.name,
+        img: workflow.item.img,
+        origin: workflow.item.uuid,
+        flags: {
+            'chris-premades': {
+                spiritOfDeathHauntCreature: {
+                    targets: Array.from(workflow.targets).map(i => i.document.uuid),
+                }
+            }
+        }
+    };
+    let casterEffect = await effectUtils.createEffect(workflow.actor, casterEffectData, {concentrationItem: workflow.item, strictlyInterdependent: true, identifier: 'spiritOfDeathHauntCreature'});
+    await genericUtils.setFlag(casterEffect, 'chris-premades', 'macros.combat', ['spiritOfDeathHauntCreatureHaunt']);
+    for (let i of workflow.targets) {
+        if (i.actor) await effectUtils.createEffect(i.actor, targetEffectData, {strictlyInterdependent: true, parentEntity: casterEffect, identifier: 'spiritOfDeathHauntCreatureHaunted'});
+    }
+    await itemUtils.createItems(workflow.actor, [featureData], {parentEntity: casterEffect});
+}
+async function turnStart({trigger: {entity: effect, token, target}}) {
+    let [targetCombatant] = game.combat.getCombatantsByToken(target.document);
+    if (!targetCombatant) return;
+    let hauntedEffect = effectUtils.getEffectByIdentifier(targetCombatant.actor, 'spiritOfDeathHauntCreatureHaunted');
```

Comment:
If I'm reading this right you don't need to get `targetCombatant` at all - can just use `target.actor` here